ray-operator/controllers/ray/utils/constant.go

```diff
@@ -191,6 +191,8 @@ const (
 
 	// KubeRayController represents the value of the default job controller
 	KubeRayController = "ray.io/kuberay-operator"
+
+	ServeConfigLRUSize = 1000
```

Comment:
LGTM. One small question is: Is it ok to re-apply serve configs due to cache evictions when we have more than 1000 active RayService?

Reply:
I think this can be handled using `ObservedGeneration`. Currently, we are not using `ObservedGeneration` to determine whether to update the CR. Once we start using it, this issue should be resolved. Perhaps we should create an issue to handle `ObservedGeneration`.

Reply:
I am not sure, but I am also afraid that 1000 is already too large, almost equivalent to the "memory leak" in the original issue. If re-applying serve configs is okay, we should probably shrink the number.

Reply:
I think it should be fine. ServeConfig is a YAML string, which is at most KB-level in size. Even with 1,000 of them, it would only be at the MB level.

Reply:
> Is it ok to re-apply serve configs

re-apply serveConfig is fine.

Ray encourages users to run multiple Serve applications in a single RayCluster. It's hard for me to imagine a user managing more than 100 RayService CRs with a single KubeRay operator.